src/internal/fl-patch.js

```diff
@@ -0,0 +1,22 @@
+var fl = require('./fl.js');
+var fixedAp = function(f) {
+  return f.ap(this);
+};
+
+var patch = function(obj){
+  return Object.keys(fl).forEach(function(key) {
+    if (typeof obj[key] === 'function') {
+      if (key === 'ap') {
+        obj[fl[key]] = fixedAp;
```

Comment:
Change the way the existing `ap` methods behave, do you mean? I think we should do so. Having an `ap` method with the old behaviour is confusing.